### PR TITLE
DSL Slack Publisher and associated grammar updates

### PR DESCRIPTION
**Pull request by jhweintraub**
## Changes:

New feature - DSL slack publisher and associated grammar updates to account for new publisher requirements.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing

The EventSourceTest.cs unit test for this publisher uses a hard-coded Webhook URL for the slack channel. Pushing this code invalidated it automatically by slack so you'll have to get the new one and replace it in the code if you wanted to test. 

https://api.slack.com/apps/A027WFC3H0A

**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Dsl/ANTLR/Interpreter.cs [hunk 1]
```diff
@@ -250,6 +252,18 @@ private void AddDiscordPublisher(string chatId)
...
 
+        private void AddSlackPublisher(string chatId)
+        {
+            var slackPublisher = new SlackPublisher(_api.EthereumJsonSerializer, chatId);
+
+            AddBlocksPublisher(slackPublisher);
+            AddTransactionsPublisher(slackPublisher);
+            AddPendingTransactionsPublisher(slackPublisher);
+            AddEventsPublisher(slackPublisher);
+
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Dsl/ANTLR/Interpreter.cs:263]: missing `AddUniswapPubslisher(slackPublisher)`

File: src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4 [hunk 2]
```diff
@@ -1,17 +1,19 @@
 grammar DslGrammar;
 
 tree: (expression)* ;
-expression : watchExpression | whereExpression | publishExpression | andCondition | orCondition ;
+expression : sourceExpression | watchExpression | whereExpression | publishExpression | andCondition | orCondition ;
+sourceExpression : SOURCE WORD ;
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4:5]: why are you introducing sourceExpression?

File: src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4 [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
-condition : WORD BOOLEAN_OPERATOR ( WORD | DIGIT | ADDRESS | BYTECODE ) ; 
+publishExpression : PUBLISH PUBLISH_VALUE ( URL | WORD | DIGIT );
+condition : WORD BOOLEAN_OPERATOR ( WORD | DIGIT | ADDRESS | BYTECODE ) ;
 andCondition : AND condition ;
 orCondition : OR condition ;
 
 BOOLEAN_OPERATOR : ARITHMETIC_SYMBOL | CONTAINS ;
 ARITHMETIC_SYMBOL : '==' | '!=' | '<' | '>' | '<=' | '>=' | IS | NOT ;
 
+SOURCE : 'SOURCE' ;
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4:16]: why is SOURCE added here?

File: src/Nethermind/Nethermind.Pipeline/Publishers/SlackPublisher.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            try {
+
+              var x = await _httpClient.PostAsync(uri, new StringContent(content, Encoding.UTF8, "application/json"));
+
+            } catch (HttpRequestException) {
+
+            }
+        }
+
+        private string LoadBotToken()
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Pipeline/Publishers/SlackPublisher.cs:97]: do we need this method??

File: src/Nethermind/Nethermind.Pipeline/Publishers/SlackPublisher.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+            var serializedData = _serializer.Serialize(data);
+
+            var headers = _httpClient.DefaultRequestHeaders;
+
+            var uri = new Uri(_chatId);
+
+            var messageContents = new FormUrlEncodedContent(new[]
+            {
+                // new KeyValuePair<string, string>("chat_id", _chatId),
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.Pipeline/Publishers/SlackPublisher.cs:80]: remove commented code

**Review by SebastianDremo — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4 (hunk 2 above)
Comment (jhweintraub) [reply src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4:5]: i'm actually not sure where that came from. May be a typo. I'll remove it.

File: src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4 (hunk 3 above)
Comment (jhweintraub) [reply src/Nethermind/Nethermind.Dsl/Grammar/DslGrammar.g4:16]: I honestly don't know, I don't remember ever writing that but I will remove it promptly

**Review by SebastianDremo — APPROVED** [review]

**Pull request merged**